source/agora/flash/Channel.d

```diff
@@ -326,10 +327,9 @@ public class Channel
 
         scope DeserializeDg dg = (size) @safe
         {
-            if (size > data.length)
-                throw new Exception(
-                    format("Requested %d bytes but only %d bytes available",
-                        size, data.length));
+            enforce(size <= data.length,
+                   "Requested {} bytes but only {} bytes available",
+                   size, data.length);
```

Comment:
This code is repeated in a few place, but I think we have a dedicated overload for this now ?